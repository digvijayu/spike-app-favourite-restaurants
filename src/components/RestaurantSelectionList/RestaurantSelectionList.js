import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ListContainer = styled.div`
  max-height: 10rem;
  scroll: auto;
`;

const ListItem = styled.div`
  padding: 0.5rem 1rem;
  &:hover {
    background: blue;
    color: white;
  }
`;

const NoItemsText = styled.div`
  color: #ccc;
  text-decoration: italic;
`;

class RestaurantSelectionList extends Component {
  handleOnItemSelect(restaurant) {
    console.log('handleOnItemSelect', restaurant);
  }

  render() {
    const { restaurants } = this.props;
    return (
      <ListContainer>
        {restaurants.length < 1 && (
          <NoItemsText>No items available.</NoItemsText>
        )}
        {restaurants.length > 0 &&
          restaurants.map(restaurant => (
            <ListItem
              key={restaurant.name}
              onClick={this.handleOnItemSelect.bind(this, restaurant)}>
              {restaurant.name}
            </ListItem>
          ))}
      </ListContainer>
    );
  }
}

RestaurantSelectionList.propTypes = {
  restaurants: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.shape({
        lat: PropTypes.number,
        lng: PropTypes.number
      })
    })
  )
};

export default RestaurantSelectionList;
