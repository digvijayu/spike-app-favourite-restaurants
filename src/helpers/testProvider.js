import React from 'react';
import { Provider } from 'react-redux';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from './../store';

configure({ adapter: new Adapter() });

function testProvider(component) {
  const { store } = configureStore();
  const jsx = (
    <Provider store={store}>{React.cloneElement(component)}</Provider>
  );
  return {
    jsx,
    store
  };
}

export default testProvider;
