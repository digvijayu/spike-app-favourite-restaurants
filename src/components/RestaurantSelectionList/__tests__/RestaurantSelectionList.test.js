import React from 'react';
import reactRenderer from 'react-test-renderer';
import 'jest-styled-components';
import RestaurantSelectionList from './../RestaurantSelectionList';
import testProvider from './../../../helpers/testProvider';

const mockRestaurantList = [
  {
    name: 'The Salt Room, 106 Kings Rd, Brighton BN1 2FU',
    location: {
      lat: 51.506552,
      lng: -0.099552
    }
  },
  {
    name: 'LEON, 7 Canvey St The Blue Fin Building, London SE1 9AN',
    location: {
      lat: 51.506552,
      lng: -0.099552
    }
  }
];

it('should render the compoenent when items are available in the list', () => {
  const { jsx } = testProvider(
    <RestaurantSelectionList restaurants={mockRestaurantList} />
  );
  const tree = reactRenderer.create(jsx).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should display no items available message when no items are available', () => {
  const { jsx } = testProvider(<RestaurantSelectionList restaurants={[]} />);
  const tree = reactRenderer.create(jsx).toJSON();
  expect(tree).toMatchSnapshot();
});
