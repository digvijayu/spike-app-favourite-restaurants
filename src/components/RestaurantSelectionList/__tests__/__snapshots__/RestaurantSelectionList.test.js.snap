// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display no items available message when no items are available 1`] = `
.c0 {
  max-height: 10rem;
  -webkit-scroll: auto;
  -moz-scroll: auto;
  -ms-scroll: auto;
  scroll: auto;
}

.c1 {
  color: #ccc;
  -webkit-text-decoration: italic;
  text-decoration: italic;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    No items available.
  </div>
</div>
`;

exports[`should render the compoenent when items are available in the list 1`] = `
.c0 {
  max-height: 10rem;
  -webkit-scroll: auto;
  -moz-scroll: auto;
  -ms-scroll: auto;
  scroll: auto;
}

.c1 {
  padding: 0.5rem 1rem;
}

.c1:hover {
  background: blue;
  color: white;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    The Salt Room, 106 Kings Rd, Brighton BN1 2FU
  </div>
  <div
    className="c1"
    onClick={[Function]}
  >
    LEON, 7 Canvey St The Blue Fin Building, London SE1 9AN
  </div>
</div>
`;
